NitroxServer/GameLogic/Spawning/EntitySpawner.cs

```diff
@@ -76,53 +70,40 @@ private IEnumerable<Entity> SpawnEntities(EntitySpawnPoint entitySpawnPoint)
                 yield break;
             }
 
-            float rollingProbabilityDensity = 0;
+            float rollingProbabilityDensity = dstData.prefabs.Sum(prefab => prefab.probability / entitySpawnPoint.Density);
 
-            PrefabData selectedPrefab = null;
-
-            foreach (PrefabData prefab in dstData.prefabs)
+            if (rollingProbabilityDensity <= 0)
             {
-                float probabilityDensity = prefab.probability / entitySpawnPoint.Density;
-                rollingProbabilityDensity += probabilityDensity;
+                yield break;
             }
 
             double randomNumber = random.NextDouble();
-            double rollingProbability = 0;
-
-            if (rollingProbabilityDensity > 0)
+            if (rollingProbabilityDensity > 1f)
```

Comment:
@Sunrunner37 Shouldn't the random number always scale to the sum of probabilities?

Reply:
Dunno - i think the algorithm was copied from UWE.

Reply:
I'll check the source code. It doesn't seem correct, but the lower probability for all items might be intended.

Reply:
It's still working like that in the game code - if the sum of prefab probabilties isn't at least `1f` this allows a SpawnPoint to not spawn anything, which is a valid usecase.